publisher.go

```diff
@@ -0,0 +1,178 @@
+package main
+
+import (
+	"fmt"
+	"time"
+	"net/http"
+	"io/ioutil"
+	"log"
+	"github.com/jlaffaye/ftp"
+	"bytes"
+	"sync"
+	"io"
+)
+
+var lock = &sync.Mutex{}
```

Comment:
Quite possibly, we're using a Python FTP server so we may not have this problem on the real implementation